r/src/compute-exec.cpp

```diff
@@ -43,6 +43,48 @@ std::shared_ptr<compute::ExecPlan> ExecPlan_create(bool use_threads) {
   return plan;
 }
 
+// [[arrow::export]]
+std::shared_ptr<compute::ExecPlan> ExecPlan_create_with_metadata(
```

Comment:
It seems rather odd to me that we would need two methods here but it may just be my ignorance of working with R.  Wouldn't it be possible to send an empty value for `metadata` if the user doesn't want to specify it?